### PR TITLE
Create `CommentPath` model and `gparent_id` column to `Comment`

### PR DESCRIPTION
**Pull request by givem2some**
## Checklist
- [X] Tested successfully.
- [X] Titles of commit message and PR are in English.
- [X] PR number at the title will be removed when `squash and merge`.
- [X] [Guideline](https://github.com/tymsai/marmoym-dev-support/blob/dev/CONTRIBUTING.md#commit-messages) is followed.

### CODE REVIEW

File: src/database/migrations/20171125180340_integrate.js [hunk 1]
```diff
@@ -94,22 +95,32 @@ exports.up = function(knex, Promise) {
       table.integer('usage_id').notNullable();
       table.string('status', 1).defaultTo('N');
```
Comment (enginehenryed) [inline src/database/migrations/20171125180340_integrate.js]: blank line

File: src/database/migrations/20171125180340_integrate.js [hunk 2]
```diff
@@ -94,22 +95,32 @@ exports.up = function(knex, Promise) {
...
+      table.increments('id').primary();
+      table.integer('parent_id').defaultTo('0');
+      table.integer('gparent_id').defaultTo('0');
+      table.string('target_type', 1).notNullable();
+      table.integer('target_id').notNullable();
+      table.string('label', 512).notNullable();
+      table.integer('user_id').notNullable();
+      table.integer('vote_id').notNullable();
+      table.string('status', 1).defaultTo('N');
+
```
Comment (enginehenryed) [inline src/database/migrations/20171125180340_integrate.js]: blank line

File: src/database/migrations/20171125180340_integrate.js [hunk 3]
```diff
@@ -94,22 +95,32 @@ exports.up = function(knex, Promise) {
...
+      table.timestamps(true, true);
+    }),
+    knex.schema.createTable('CommentPath', function(table) {
+      table.increments('id').primary();
+      table.integer('gparent_id').defaultTo('0');
+      table.string('path', 1024);
+      table.string('status', 1).defaultTo('N');
+
```
Comment (enginehenryed) [inline src/database/migrations/20171125180340_integrate.js]: blank line

File: src/database/migrations/20171125180340_integrate.js [hunk 4]
```diff
@@ -94,22 +95,32 @@ exports.up = function(knex, Promise) {
...
-    knex.schema.dropTable('VoteRecord'),
-    knex.schema.dropTable('Origin'),
-    knex.schema.dropTable('Pos'),
-    knex.schema.dropTable('DefinitionPos'),
-    knex.schema.dropTable('Usage'),
-    knex.schema.dropTable('DefinitionUsage')
-  ]);  
-};
\ No newline at end of file
+
```
Comment (enginehenryed) [inline src/database/migrations/20171125180340_integrate.js]: blank line

File: src/database/migrations/20171125180340_integrate.js (hunk 1 above)
Comment (UnoRich) [reply src/database/migrations/20171125180340_integrate.js]: i agree

**Pull request merged**